packages/core/src/lib/store/index.ts

```diff
@@ -211,53 +212,67 @@ class Store extends BaseProtocol implements IStore {
     decoders: IDecoder<T>[],
     options?: QueryOptions
   ): AsyncGenerator<Promise<T | undefined>[]> {
-    const { pubSubTopic = DefaultPubSubTopic } = this.options;
-
     let startTime, endTime;
 
     if (options?.timeFilter) {
       startTime = options.timeFilter.startTime;
       endTime = options.timeFilter.endTime;
     }
 
-    const decodersAsMap = new Map();
-    decoders.forEach((dec) => {
-      if (decodersAsMap.has(dec.contentTopic)) {
-        throw new Error(
-          "API does not support different decoder per content topic"
-        );
-      }
-      decodersAsMap.set(dec.contentTopic, dec);
-    });
-
-    const contentTopics = decoders.map((dec) => dec.contentTopic);
-
-    const queryOpts = Object.assign(
-      {
-        pubSubTopic: pubSubTopic,
-        pageDirection: PageDirection.BACKWARD,
-        pageSize: DefaultPageSize
-      },
-      options,
-      { contentTopics, startTime, endTime }
+    const _pubSubTopics = new Set(
+      decoders.map((decoder) => decoder.pubSubTopic)
```

Comment:
I wonder if it'd be fine to assume that the store query is for one pubsub topic and reject to user if several pubsub topics are passed in one query.

The issue is that underneath, you are now triggering several queries instead of just one which makes error handling more complex.